K8s usage guide (#685)
ngrok/ngrok-docs: docs/k8s/user-guide.mdx

                port:
                  number: <service-port>
```

> # TCP and TLS Edges {#tcp-tls-edges}

Reviewer: I wasn't seeing this in the TOC above.

Author: @OfTheDelmer This should be resolved now. Some of the headings were not set correctly.

Policies](https://ngrok.com/docs/tls/ip-restrictions/) for limiting access
to the edge. At the time of writing, these policies must be provided as a
reference in the form `ipp_<id>`.

On top of the options available to TCP Edges, TLS Edges support (and require) a few other options:

- (required) `hostports`: A list of `"<fqdn>:443"` strings declaring the list of
  reserved domains for the edge to listen on.
- [`tlsTermination`](https://ngrok.com/docs/api/resources/tls-edge-tls-termination-module/): Configure the TLS Termination behavior. The `terminateAt` field may be set to `upstream` to pass the encrypted stream to the Tunnel backend, or `edge` to terminate the TLS stream at the ngrok edge, and pass plaintext bytes to the Tunnel.
- [`mutualTls`](https://ngrok.com/docs/api/resources/tls-edge-mutual-tls-module/): Configure client certificate validation at the edge. Requires a reference to a [Certificate Authority](https://ngrok.com/docs/api/resources/certificate-authorities/).

TCP Example:

```yaml
apiVersion: ingress.k8s.ngrok.com/v1alpha1
kind: TCPEdge
metadata:
  name: test-edge
spec:
  backend:
    labels:
      app: tcptestedge
```

> Because TCP Edges don't currently support providing a reserved TCP address. On edge creation, one will be allocated for them, and will be visible by checking the status of the resource:

Reviewer: ```suggestion
TCP Edges currently do not support providing a reserved TCP address. Therefore, one will be allocated upon edge creation. This can be viewed by checking the status of the resource:
```
